extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java

```diff
@@ -104,36 +104,70 @@ public Result<Map<String, Object>> getVerifiedCredentials(DidDocument didDocumen
 
         var verifiedCredentials = verifyCredentials(verifiableCredentials, didDocument);
 
-        monitor.debug(() -> String.format("Verified %s credentials", verifiedCredentials.size()));
+        monitor.debug(() -> String.format("Verified %s credentials", verifiedCredentials.getContent().size()));
 
-        var claims = extractClaimsFromCredential(verifiedCredentials);
+        var claims = extractClaimsFromCredential(verifiedCredentials.getContent());
 
-        if (claims.failed()) {
+        var failureMessages = Stream.concat(verifiedCredentials.getFailureMessages().stream(),
+                claims.getFailureMessages().stream()).collect(Collectors.toList());
+
+        var result = new VerificationResult<>(claims.getContent(), failureMessages);
+
+        if (result.failed()) {
```

Comment:
I would add a comment at this point explaining that this block is a temporary solution until the API is changed to support a Result with both content and failures, so that the client can decide how to proceed.